impl/src/main/java/jakarta/faces/component/ActionSource2.java

```diff
@@ -26,7 +26,9 @@
  * </p>
  *
  * @since 1.2
+ * @deprecated Use {@link ActionSource} instead.
  */
+@Deprecated(since = "4.1", forRemoval = true)
```

Comment:
I doubt a 4.1 release will be done. Can you retarget it for 5.0?

Reply:
Will do when 4.1 will be merged into 5.0 because it's easier to find&replace 4.1 by 5.0 in one go.